sdk/cosmos/src/clients/cosmos_client.rs

```diff
@@ -54,21 +48,37 @@ impl CosmosOptions {
 }
 
 /// Create a Pipeline from CosmosOptions
-fn new_pipeline_from_options(options: CosmosOptions) -> Pipeline {
+fn new_pipeline_from_options(
+    options: CosmosOptions,
+    authorization_token: AuthorizationToken,
+) -> Pipeline<CosmosContext> {
+    let auth_policy: Arc<dyn azure_core::Policy<CosmosContext>> =
+        Arc::new(crate::AuthorizationPolicy::new(authorization_token));
+
+    let mut per_retry_policies = Vec::with_capacity(1);
```

Comment:
This call to `with_capacity` doesn't really help. The call to push would reserve space on the heap anyway, and `with_capacity` doesn't reserve exact capacity anyway.